contracts/Disputer.sol

```diff
@@ -0,0 +1,106 @@
+// SPDX-License-Identifier: Apache-2.0
+pragma solidity 0.8.24;
+
+import { IERC20 } from "@openzeppelin/contracts/token/ERC20/IERC20.sol";
+import { Ownable } from "@openzeppelin/contracts/access/Ownable.sol";
+import { Distributor } from "./Distributor.sol";
+
+contract Disputer is Ownable {
+    Distributor public distributor;
+    mapping(address => bool) public whitelist;
+
+    // ================================= ERRORS =================================
+
+    error NotWhitelisted();
+    error DisputeFundsTransferFailed();
+    error EthNotAccepted();
+    error UnresolvedDispute();
+    error WithdrawalFailed();
+    // ================================= MODIFIERS =================================
+
+    /// @notice Checks whether the `msg.sender` is a whitelisted address
+    modifier onlyWhitelisted() {
+        if (!whitelist[msg.sender]) revert NotWhitelisted();
+        _;
+    }
+
+    // ================================ CONSTRUCTOR ================================
```

Comment:
selling my own church but to be uniform with the rest you could setup https://github.com/Picodes/headers :P